pkg/controller/seed-controller-manager/mla/mla.go

```diff
@@ -102,20 +107,40 @@ func Add(
-		return fmt.Errorf("failed to create mla project controller: %w", err)
-	}
-	if err := newOrgUserGrafanaReconciler(mgr, log, numWorkers, workerName, versions, grafanaClient, mlaEnabled); err != nil {
-		return fmt.Errorf("failed to create mla userprojectbinding controller: %w", err)
-	}
-	if err := newDatasourceGrafanaReconciler(mgr, log, numWorkers, workerName, versions, httpClient, grafanaURL, grafanaAuth, mlaNamespace, overwriteRegistry, mlaEnabled); err != nil {
-		return fmt.Errorf("failed to create mla cluster controller: %w", err)
-	}
-	if err := newAlertmanagerReconciler(mgr, log, numWorkers, workerName, versions, httpClient, cortexAlertmanagerURL, mlaEnabled); err != nil {
-		return fmt.Errorf("failed to create mla alertmanager configuration controller: %w", err)
-	}
-	if err := newUserGrafanaReconciler(mgr, log, numWorkers, workerName, versions, grafanaClient, httpClient, grafanaURL, grafanaHeader, mlaEnabled); err != nil {
-		return fmt.Errorf("failed to create mla user controller: %w", err)
+	orgGrafanaController := newOrgGrafanaController(mgr.GetClient(), log, grafanaClient)
+	alertmanagerController := newAlertmanagerController(mgr.GetClient(), log, httpClient, cortexAlertmanagerURL)
+	orgUserGrafanaController := newOrgUserGrafanaController(mgr.GetClient(), log, grafanaClient)
+	datasourceGrafanaController := newDatasourceGrafanaController(mgr.GetClient(), httpClient, grafanaURL, grafanaAuth, mlaNamespace, log, overwriteRegistry)
+	userGrafanaController := newUserGrafanaController(mgr.GetClient(), log, grafanaClient, httpClient, grafanaURL, grafanaHeader)
+	if mlaEnabled {
+		if err := newOrgGrafanaReconciler(mgr, log, numWorkers, workerName, versions, orgGrafanaController); err != nil {
+			return fmt.Errorf("failed to create mla project controller: %w", err)
+		}
+		if err := newOrgUserGrafanaReconciler(mgr, log, numWorkers, workerName, versions, orgUserGrafanaController); err != nil {
+			return fmt.Errorf("failed to create mla userprojectbinding controller: %w", err)
+		}
+		if err := newDatasourceGrafanaReconciler(mgr, log, numWorkers, workerName, versions, datasourceGrafanaController); err != nil {
+			return fmt.Errorf("failed to create mla cluster controller: %w", err)
+		}
+		if err := newAlertmanagerReconciler(mgr, log, numWorkers, workerName, versions, alertmanagerController); err != nil {
+			return fmt.Errorf("failed to create mla alertmanager configuration controller: %w", err)
+		}
+		if err := newUserGrafanaReconciler(mgr, log, numWorkers, workerName, versions, userGrafanaController); err != nil {
+			return fmt.Errorf("failed to create mla user controller: %w", err)
+		}
+	} else {
+		cleanupController := newCleanupController(
+			mgr.GetClient(),
+			log,
+			datasourceGrafanaController,
+			alertmanagerController,
+			orgUserGrafanaController,
+			orgGrafanaController,
+			userGrafanaController,
+		)
+		if err := newCleanupReconciler(mgr, log, numWorkers, workerName, versions, cleanupController); err != nil {
```

Comment:
Can you please add a comment about this cleanup reconciler to the `Add` function?

Reply:
sure.
done